test/timeout/timeout_test.go

```diff
@@ -87,5 +85,5 @@ func checkTimeout(t *testing.T, client *http.Client, name string, code int, init
 		t.Fatal(err)
 	}
 	defer resp.Body.Close()
-	assert.Equal(t, resp.StatusCode, code)
+	assert.True(t, resp.StatusCode == http.StatusRequestTimeout || resp.StatusCode == http.StatusGatewayTimeout)
```

Comment:
The status code was changed by Envoy upstream - https://github.com/knative-sandbox/net-kourier/issues/1015#issuecomment-1542258326

I considered to use `KOURIER_GATEWAY_IMAGE` to distinguish the version but it is a minor stuff so I think it is alright for test with "either" match.